_posts/2020-02-14-creating-doc-assemblies.md

```diff
@@ -0,0 +1,180 @@
+If that’s too much to absorb,
+I’m sure you’ll quickly get the gist by copying what you see in the Strimzi files.
+
+You can use any text editor to work with Asciidoc,
+but if you are on the hunt for a decent editor geared to working with Asciidoc, I think [Atom](https://atom.io/) takes some beating.
+
+You can use [Asciidoctor](https://asciidoctor.org/) to build the Strimzi guides locally, or you can use the [make tooling supplied with the documentation](https://strimzi.io/contributing/guide/#make-tooling).
+
+### Monolith to modular
+
+If you glance at the Strimzi [documentation folder](https://github.com/strimzi/strimzi-kafka-operator/tree/master/documentation), you’ll be unsurprised to see that the Strimzi documentation isn’t written in a single file.
+There’s some structure there, and not by accident.
+Thrive in structure or drown in chaos, as they say.
+
+However, it might not be immediately noticeable what the structure represents.
+Essentially, it’s a reflection of the modular approach taken with the documentation.
+You will see folders for guides (using, quickstart, overview), and folders for the files that provide the content for one or more of these guides -- assemblies and modules.
+
+Modularization separates concepts from tasks, then wraps them into an assembly.
+An assembly represents a user story.
+The assembly should answer a question -- the what, why and how of doing something specific.
+As such, the assembly represents a self-contained unit of content that should be understood in isolation.
+
+If you want to read more on modularization of documentation,
+take a look at this [reference guide](https://redhat-documentation.github.io/modular-docs/).
+
+## Creating a new documentation assembly
+
+Okay, so how do you go about creating a new assembly?
+Let’s take a look at a specific assembly I added to the Strimzi documentation to describe support for OAuth 2.0.
+I worked on the assembly locally, and then pushed the content to the Strimzi repo as a merge request.
+
+A new assembly requires:
+
+1. [An assembly file](https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/assemblies/oauth/assembly-oauth.adoc)
+2. [Modules to contain the content](https://github.com/strimzi/strimzi-kafka-operator/tree/master/documentation/modules/oauth)
+
+So, the first thing I did was create the assembly (assembly-oauth.adoc) that would house the concepts and tasks related to OAuth 2.0.
+The assembly [describes how to use OAuth 2.0 token-based authentication](https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/assemblies/oauth/assembly-oauth.adoc).
+This sits in its own oauth category folder under the assemblies folder.
+Categories make it easier to find related material.
+
+````
+documentation
+└── assemblies
+      └── oauth
+            └── assembly-oauth.adoc
+````
+
+Assemblies are kept separate from the modules that form the content -- the concepts, procedures, and reference files.
```

Comment:
I think we should mention somewhere what are the different modules: concept, procedure and reference.

Reply:
> Assemblies are kept separate from the modules that form the content.
> Modules can include _concepts_ to provide understanding, _procedures_ to describe the steps to accomplish certain tasks, and related _reference_ material, often in lists or tables.